controller/kubernetes_pod_controller.go

```diff
@@ -199,6 +200,28 @@ func (kc *KubernetesPodController) handlePodDeletionIfNodeDown(pod *v1.Pod, node
 		return nil
 	}
 
+	// make sure the volumeattachments of the pods are gone first
+	// ref: https://github.com/longhorn/longhorn/issues/2947
+	vas, err := kc.getVolumeAttachmentsOfPod(pod)
+	if err != nil {
+		return err
+	}
+	for _, va := range vas {
+		if va.DeletionTimestamp == nil {
+			err := kc.kubeClient.StorageV1().VolumeAttachments().Delete(context.TODO(), va.Name, metav1.DeleteOptions{})
+			if err != nil {
+				if datastore.ErrorIsNotFound(err) {
+					continue
+				}
+				return err
+			}
+			kc.logger.Infof("%v: deleted volume attachment %v for pod %v on downed node %v", controllerAgentName, va.Name, pod.Name, nodeID)
+			return err
+		}
+		// wait the volumeattachment object to be deleted
```

Comment:
Probably we can have a log here to know the pod is not deleted yet because the VAs are still under deletion.

Reply:
Thank you. Fixed